ppinot-model/src/main/java/es/us/isa/ppinot/evaluation/logs/XESLog.java

```diff
@@ -0,0 +1,231 @@
+	            	}
+                }else {
+	            	lastLogEntry = createLogEntry(processId, instanceId, instanceData);
+	            	updateListeners(lastLogEntry);
+	            	break;
+                }
+            }
+        } catch (XMLStreamException e) {
+        	log.log(Level.WARNING, "Error processsing entry: " + reader.getLocation(), e);
+        }
+    }
+    
+
+    private LogEntry createLogEntry(String processId, String instanceId, Map<String, Object> instanceData) {
+        boolean endEntry = false;
+        String bpElement = "";
+        LogEntry.EventType eventType = null;
+        String originator = null;
+        DateTime timeStamp = null;
+        DateTimeFormatter fmt = ISODateTimeFormat.dateTime();
+        Map<String, Object> data = new HashMap<String,Object>();
+
+        try {
+            while (reader.hasNext() && !endEntry) {
+                int type = reader.next();
+                
+                if (isEndElement(type) && hasName(reader, TAG_EVENT)) {
+                    endEntry = true;
+                } else if (isStartElement(type) && hasName(reader, TAG_LOG)) {
+                	String att = reader.getAttributeValue(0);             	
+                	if(att.equals(ATTRIBUTE_ID)) {
+                		bpElement = reader.getAttributeValue(1);
+                		
+                	}else if(att.equals(TAG_EVENTTYPE)){
+                		eventType = LogEntry.EventType.valueOf(reader.getAttributeValue(1));
+                		
+                	}else if(att.equals(TAG_ORIGINATOR)) {
+                		originator = reader.getAttributeValue(1);
+                		
+                	}else {
+                		data = createData(data);
+                	}
+                	reader.next();
+                } else if (isStartElement(type) && hasName(reader, TAG_TIME)) {
+                	if(reader.getAttributeValue(0).equals(TAG_TIMESTAMP)) {
+                		timeStamp = fmt.parseDateTime(reader.getAttributeValue(1));
+                		reader.next();
+                	}
+                } 
+            }
+        } catch (XMLStreamException e) {
+            log.log(Level.WARNING, "Error processsing entry: " + reader.getLocation(), e);
+        } catch (Exception e) {
+        	log.log(Level.WARNING, "Error processsing entry: " + reader.getLocation(), e);
+        }
+
+        LogEntry entry = LogEntry.flowElement(processId, instanceId, bpElement, eventType, timeStamp);
+        
+        if (instanceData != null) {
+            entry = entry.withData(instanceData);
+        }
+        
+        if (data != null) {
+            entry = entry.withData(data);
+        }
+
+        if (originator != null)
+            entry.setResource(originator);
+       
+        return entry;
+    }
+
+    private Map<String, Object> createData(Map<String, Object> data) {
+    
+        try {
+        	
+            String key = reader.getAttributeValue(0);
+            String value = reader.getAttributeValue(1);
```

Comment:
Since the element already provides the type of element (date, string, int, etc.) It would be great if it performs the conversion already here so that value instead of a String is of the right type. 

For instance, if the data is: `<boolean key="paid" value="true"/>`
The data created should be a boolean with value true instead of a string with value "true".